bundle/terranova/tnresources/adapter.go

```diff
@@ -47,20 +72,36 @@ type IResource interface {
 
 	// [Optional] WaitAfterUpdate waits for the resource to become ready after update.
 	WaitAfterUpdate(ctx context.Context, config any) error
+}
 
-	// [Optional] RecreateFields returns a list of fields that will cause resource recreation if changed
-	RecreateFields() []string
+// Alternative methods that also return remote state.
+// Only use these if you get remote state for free as part of the main operation. Otherwise, prefer simpler NoRefresh variants.
+type IResourceAdapterWithRefresh interface {
```

Comment:
The name suggests it contains a refresh function but I don't see the `DoRefresh` function below.